aata.py

```diff
@@ -39,136 +39,162 @@
 			MakeLinkedArtOrganization
 from settings import *
 
-# Set up environment
-def get_services(**kwargs):
-	return {
-		'trace_counter': itertools.count(),
-		'gpi': create_engine(gpi_engine),
-		'aat': create_engine(aat_engine),
-		'uuid_cache': create_engine(uuid_cache_engine),
-		'fs.data.aata': bonobo.open_fs(aata_data_path)
-	}
-
 ### Pipeline
 
-if DEBUG:
-	sys.stderr.write("In DEBUGGING mode\n")
-	LIMIT		= int(os.environ.get('GETTY_PIPELINE_LIMIT', 10))
-	PACK_SIZE	= 10
-	SRLZ		= Serializer(compact=False)
-	WRITER		= FileWriter(directory=output_file_path)
-	# WRITER	= ArchesWriter()
-else:
-	LIMIT		= 10000000
-	PACK_SIZE	= 10000000
-	SRLZ		= Serializer(compact=True)
-	WRITER		= FileWriter(directory=output_file_path)
-	# WRITER	= ArchesWriter()
-
-
 class AddDataDependentArchesModel(Configurable):
 	models = Option()
 	def __call__(self, data):
 		data['_ARCHES_MODEL'] = self.models['LinguisticObject']
 		return data
 
-def add_articles_chain(graph, records):
-	articles = graph.add_chain(
-		Limit(LIMIT),
-		make_aata_article_dict,
-		add_uuid,
-		add_aata_object_type,
-		MakeLinkedArtLinguisticObject(),
-		AddDataDependentArchesModel(models=arches_models),
-		_input=records.output
-	)
-	if True:
-		# write ARTICLES data
-		graph.add_chain(
-			SRLZ,
-			WRITER,
+class AATAPipeline:
+	def __init__(self, input_path, files, limit=None, debug=False):
+		self.files = files
+		self.limit = limit
+		self.debug = debug
+		self.input_path = input_path
+		if self.debug:
+			self.files = [self.files[0]]
+			self.serializer	= Serializer(compact=False)
+			self.writer		= None
+			# self.writer	= ArchesWriter()
+			sys.stderr.write("In DEBUGGING mode\n")
+		else:
+			self.serializer	= Serializer(compact=True)
+			self.writer		= None
+			# self.writer	= ArchesWriter()
+
+	# Set up environment
+	def get_services(self, **kwargs):
+		return {
+			'trace_counter': itertools.count(),
+			'gpi': create_engine(gpi_engine),
+			'aat': create_engine(aat_engine),
+			'uuid_cache': create_engine(uuid_cache_engine),
+			'fs.data.aata': bonobo.open_fs(self.input_path)
+		}
+
+	def add_serialization_chain(self, graph, input):
+		if self.writer is not None:
+			graph.add_chain(
+				self.serializer,
+				self.writer,
+				_input=input
+			)
+		else:
+			sys.stderr.write('*** No serialization chain defined\n')
+
+	def add_articles_chain(self, graph, records):
+		if self.limit is not None:
+			records = graph.add_chain(
+				Limit(self.limit),
+				_input=records.output
+			)
+		articles = graph.add_chain(
+			make_aata_article_dict,
```

Comment:
If we're building a pipeline class per project, should the transformation functions be members of the class?

Reply:
Possibly, though I'm wary of turning this into either one huge class for the entire project, or a deep class hierarchy. I'll try to circle back around after the code stabilizes a bit and see what could make sense being part of the project class.